xclim/core/dataflags.py

```diff
@@ -718,7 +723,7 @@ def ecad_compliant(
 
     Returns
     -------
-    Union[xarray.DataArray, xarray.Dataset]
+    xarray.DataArray or xarray.Dataset or None
```

Comment:
Never heard of it, but it would be interesting to see what it gives us. I'll open an issue about it, so we can both look at it and coordinate our efforts.

At one point in time, I wanted to us to have our code fully `mypy` compatible, but it was far too much work at the time. We're much closer to that goal now. This would also be worth looking into.